R/install-git.r

```diff
@@ -69,3 +76,52 @@ remote_metadata.git_remote <- function(x, bundle = NULL, source = NULL) {
     RemoteSha = sha
   )
 }
+
+#' @export
+remote_package_name.git_remote <- function(remote, ...) {
+
+  tmp <- tempfile()
+  on.exit(unlink(tmp))
+  description_path <- paste0(collapse = "/", c(remote$subdir, "DESCRIPTION"))
+
+  # Try using git archive --remote to retrieve the DESCRIPTION, if the protocol
+  # or server doesn't support that return NULL
+  res <- try(silent = TRUE,
```

Comment:
Theoretically you probably could, but it would require a patch to git2r, possibly . Using `git archive --remote` is really special case and we are using it as a hack to retrieve just a specific file from a git repository. There isn't really a standard way to do so normally with git.

So while it is probably possible to do I am not sure it is worth the time investment to add it to git2r. We have ways to get the file for the other remotes, just the raw git one needs to use this functionality so I think it is worth relying on the user having a git client installed. If they do not the behavior will be the same as the current behavior, which seems to be OK for most people.